code/_core/obj/projectile/magic.dm

```diff
@@ -2,6 +2,77 @@
 	icon = 'icons/obj/projectiles/magic.dmi'
 
 	collision_bullet_flags = FLAG_COLLISION_BULLET_LIGHT
+	//Do go boom?
+	var/does_explode = FALSE
+	var/explode_power = 5
+	//Do we slowly fade out of existance?
+	var/does_degrade = FALSE
+	var/magic_vel_degrade = 0.90 
```

Comment:
These two vars can be combined. Just use magic_vel_degrade  and check if magic_vel_degrade > 0 for does_degrade usage.

Reply:
combined.